src/plugins/intel_cpu/src/nodes/topk.cpp

```diff
@@ -2041,14 +2041,12 @@ void TopK::preset_params() {
     }
 
     if (isDynamicNode()) {
-        if (stable) {
+        if (stable ||
+            !((layout == TopKLayoutType::topk_ncsp || layout == TopKLayoutType::topk_nspc) && topk_innermost)) {
```

Comment:
could you please use some named variable for the combination of the checks.
I.e.
const bool canUseHeapSort = (layout == TopKLayoutType::topk_ncsp || layout == TopKLayoutType::topk_nspc) && topk_innermost

Reply:
Done